src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -211,15 +211,19 @@ CHIP_ERROR DescriptorAttrAccess::ReadDeviceAttribute(EndpointId endpoint, Attrib
 
     return err;
 }
-
+#ifdef CONFIG_USE_ENDPOINT_UNIQUE_ID
```

Comment:
The way config defines work normally is:

1. They start with CHIP_CONFIG_
2. They are defined to some default in a config header that is guaranteed to include the config headers that applications would then use to override them
3. They are tested with `#if`.

Concretely here: what config header is this file including that guarantees that it will see this define if it's defined?  With the `#if` approach, the compiler will catch that for you, at least with some warning/error flags we use on some platforms.